libkbfs/key_manager.go

```diff
@@ -653,6 +710,20 @@ func (km *KeyManagerStandard) Rekey(ctx context.Context, md *RootMetadata, promp
 	km.config.Reporter().Notify(ctx, rekeyNotification(ctx, km.config, resolvedHandle,
 		false))
 
+	// Save the previous TLF crypt key. It's symmetrically encrypted and appened to a list
+	// for MDv3 metadata.
+	var prevTlfCryptKey TLFCryptKey
+	if currKeyGen >= KeyGen(FirstValidKeyGen) && !md.AreHistoricTLFCryptKeysPerDeviceEncrypted() {
```

Comment:
Ah, I see you need `AreHistoric...` here, too. Hmm. What about making `finalizeRekey` for `V3` return an error if `prevTLFCryptKey` is nil and it's not the first keygen? And then this code can check for that error and retrieve the current crypt key and retry. That seems clearer than having to know that `prevTlfCryptKey` can be nil for v2.